corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -240,9 +254,13 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
                 answers = {},
                 model = this.parentModel;
             $fields.each(function (index) {
-                if (this.value !== '') {
-                    answers[model[index].get('id')] = encodeValue(model[index], this.value);
+                if (this.tagName === 'SELECT' && this.multiple) {
```

Comment:
I think you could simplify this block to just `var val = $(this).value()` - the value function will return `this.value` for non-selects.

Reply:
Given `this` is the JS DOM object `$(this).value()` is an invalid function call. You probably meant `$(this).val()`

But I am not sure if that works well, the reason I had to do this here was that for multiselects `$(this).val()` would return a list of selected options while `this.value` would return only first selected option. Previously `this.value` worked for all, but now since it doesn't, I used `$(this).val()` for the case of multiselects

Reply:
Oops, yes, I meant `$(this).val()`

> Previously this.value worked for all, but now since it doesn't, I used $(this).val() for the case of multiselects

The part I don't understand is, why doesn't `$(this).val()` work for all?

Reply:
The reason is for single selects `$(this).val()` will be a string, for multiselects it will be a list of strings.